postgresql/postgresql_plugin.cc

```diff
@@ -0,0 +1,832 @@
+        if (PQresultStatus(res) != PGRES_COMMAND_OK) {
+            string errorMsg(PQresultErrorMessage(res));
+            PQclear(res);
+            PQfinish(conn);
+            throw HttpReturnException(400, "Could not insert data in PostgreSQL table:  ", errorMsg);
+        }
+
+        PQclear(res);
+    }
+
+    virtual void recordRowItl(const RowName & rowName,
+                              const std::vector<std::tuple<ColumnName, CellValue, Date> > & vals) override
+    {
+        auto conn = startConnection();
+
+        insertRowPostgresql(conn, vals);
+
+        PQfinish(conn);
+    }
+    
+    virtual void recordRows(const std::vector<std::pair<RowName, std::vector<std::tuple<ColumnName, CellValue, Date> > > > & rows) override
+    {
+        auto conn = startConnection();
+
+        for (auto& vals : rows) {
+            insertRowPostgresql(conn, vals.second);
+        }        
+
+        PQfinish(conn);
+    }
+
+    /** Commit changes to the database.  Default is a no-op. */
+    virtual void commit() override
+    {
+    }
+
+    virtual std::pair<Date, Date> getTimestampRange() const override
+    {
+        throw HttpReturnException(400, "Postgresql recorder dataset is record-only");
+    }
+
+    virtual Date quantizeTimestamp(Date timestamp) const override
+    {
+        throw HttpReturnException(400, "Postgresql recorder dataset is record-only");
+    }
+
+    virtual std::shared_ptr<MatrixView> getMatrixView() const override
+    {
+        throw HttpReturnException(400, "Postgresql recorder dataset is record-only");
+    }
+
+    virtual std::shared_ptr<ColumnIndex> getColumnIndex() const override
+    {
+        throw HttpReturnException(400, "Postgresql recorder dataset is record-only");
+    }
+
+    virtual std::shared_ptr<RowStream> getRowStream() const override
+    {
+        throw HttpReturnException(400, "Postgresql recorder dataset is record-only");
+    }    
+
+};
+
+
+/*****************************************************************************/
+/* POSTGRESQL IMPORT PROCEDURE                                               */
+/*****************************************************************************/
+
+struct PostgresqlImportConfig: public ProcedureConfig {
+
+    static constexpr const char * name = "postgresql.import";
+
+    string databaseName;
+    int port;
+    string host;
+    string postgresqlQuery;
+
+    /// The output dataset.  Rows will be dumped into here via insertRows.
+    PolyConfigT<Dataset> outputDataset;
+
+    PostgresqlImportConfig() {
+        databaseName = "database";
+        port = 1234;
+        host = "localhost";
+        postgresqlQuery = "";
+
+        outputDataset.withType("sparse.mutable");
+    }
+
+};
+
+DECLARE_STRUCTURE_DESCRIPTION(PostgresqlImportConfig);
+DEFINE_STRUCTURE_DESCRIPTION(PostgresqlImportConfig);
+
+PostgresqlImportConfigDescription::
+PostgresqlImportConfigDescription()
+{
+    addParent<ProcedureConfig>();
```

Comment:
The addParent should be done at the end if I recall correctly so that the runOnCreation flag appear last like with all other procedures.